pkg/downloader/chart_downloader.go

```diff
@@ -72,6 +72,25 @@ type ChartDownloader struct {
 	RepositoryCache  string
 }
 
+// atomicWriteFile atomically (as atomic as os.Rename allows) writes a file to a
+// disk.
+func atomicWriteFile(filename string, body []byte, mode os.FileMode) error {
+	tempFile, err := ioutil.TempFile(filepath.Split(filename))
+	if err != nil {
+		return err
+	}
+	tempName := tempFile.Name()
+	tempFile.Close()
```

Comment:
why are we closing the file, then re-opening on line 85 with `ioutil.WriteFile`? Those operations are (relatively) expensive. Can't we just write to the file directly once we have the handle? Should be able to just call `io.Copy(tempFile, bytes.NewBuffer(body))` to copy the body to the file, then call `tempFile.Close()`.

Reply:
Done